cli/azd/cmd/deploy.go

```diff
@@ -22,84 +23,116 @@ import (
 	"github.com/spf13/pflag"
 )
 
-func deployCmd(rootOptions *internal.GlobalCommandOptions) *cobra.Command {
-	cmd := commands.Build(
-		&deployAction{rootOptions: rootOptions},
-		rootOptions,
-		"deploy",
-		"Deploy the application's code to Azure.",
-		&commands.BuildOptions{
-			//nolint:lll
-			Long: `Deploy the application's code to Azure.
+type deployFlags struct {
+	serviceName  string
+	outputFormat *string // pointer to allow delay-initialization when used in "azd up"
+	global       *internal.GlobalCommandOptions
+}
+
+func (d *deployFlags) Bind(local *pflag.FlagSet, global *internal.GlobalCommandOptions) {
+	d.bind(local, global)
+
+	d.outputFormat = convert.RefOf("")
+	output.AddOutputFlag(
+		local,
+		d.outputFormat,
+		[]output.Format{output.JsonFormat, output.NoneFormat},
+		output.NoneFormat)
+}
+
+func (d *deployFlags) bind(local *pflag.FlagSet, global *internal.GlobalCommandOptions) {
```

Comment:
Ahh - I understand now - I think this is done because you need want to register these flags in the `up` command as well, but you don't want to call `output.AddOutputFlag` multiple times.

Maybe name this `Bind` (since other things call it) and the other `bindAndAddOutput` or something?

Or perhaps just a comment on `bind` explaining what's going on and in what cases others should call it?